### DISCUSSION TITLE
How to use Schemathesis without relying on `pytest`?

### DISCUSSION THREAD

**Discussion by geeeez** [schemathesis/schemathesis#1439] [Q&A]
How to write Python code and run it directly instead of relying on pytest to call the written demo on the command line。
Whether schema provides such an interface, which can be run directly through Python code rather than tested with pytest on the command line.
<img width="758" alt="image" src="https://user-images.githubusercontent.com/29922543/153828123-82db4059-7e9b-415e-836a-9e8c521e29b1.png">

**Comment by Stranger6667** [comment]
Hello @geeeez !

I think you might want to use [Schemathesis Runner](https://schemathesis.readthedocs.io/en/stable/extending.html#schemathesis-test-runner) for this purpose

```python
import schemathesis

schema = schemathesis.from_uri("http://127.0.0.1:8080/swagger.json")

runner = schemathesis.runner.from_schema(schema)
for event in runner.execute():
    ...  # do something with event
```

Though, you'll need to handle [events](https://github.com/schemathesis/schemathesis/blob/master/src/schemathesis/runner/events.py) yourself

**Comment by geeeez** [comment]
Yes,thanks.
